### DISCUSSION TITLE
What are the requirements for Persistors to reach stable?

### DISCUSSION THREAD

**Discussion by richraid21** [TanStack/query#2520] [General]
Hi, I'm wondering what outstanding work is required (or believed) to be necessary before declaring the Persistence features to be stable & ready for production?

**Comment by TkDodo** [comment]
I don't know, I think it _is_ production ready, we just might make breaking changes to the api. We recently did that with the localStoragePersistor when it was renamed to webStoragePersistor

**Reply by richraid21** [reply]
It's great to hear that the desired functionality is there, but it's difficult for my teams to take it on as a dependency when the API may not be stable. 

Could we begin to drive a discussion that future breaking changes will be held for major version changes?

**Reply by TkDodo** [reply]
you can always suggest to your teams to create a lightweight abstraction over the experimental peristors where you are in control over the interface and can keep the API stable even if we change our api. If an update is necessary, it would be confined to one place and likely easy to do (like with the aforementioned rename).
